editor/components/post-featured-image/index.js

```diff
@@ -15,14 +15,19 @@ import { MediaUploadButton } from '@wordpress/blocks';
  * Internal dependencies
  */
 import './style.scss';
-import { getEditedPostAttribute } from '../../selectors';
+import { getCurrentPostType, getEditedPostAttribute } from '../../selectors';
 import { editPost } from '../../actions';
 
-function PostFeaturedImage( { featuredImageId, onUpdateImage, onRemoveImage, media } ) {
+function PostFeaturedImage( { featuredImageId, onUpdateImage, onRemoveImage, media, postType } ) {
+	if ( ! postType || ! postType.data ) {
```

Comment:
Do we really need to be so extreme as to hide the entire component when waiting for the label to become known? Can we show a temporary one, or is the text changing considered more jarring than the element suddenly becoming visible on the page?